cabal-install-solver/src/Distribution/Solver/Modular/MessageUtils.hs

```diff
@@ -0,0 +1,57 @@
+-- | Utility functions providing extra context to cabal error messages
+
+module Distribution.Solver.Modular.MessageUtils (
+  allKnownExtensions,
+  cutoffRange,
+  mostSimilarElement,
+  showUnsupportedExtension,
+  showUnsupportedLanguage,
+  withinRange
+) where
+
+import Data.Foldable (minimumBy)
+import Data.Ord (comparing)
+import Distribution.Pretty (prettyShow) -- from Cabal
+import Language.Haskell.Extension
+         ( Extension(..), Language(..), knownLanguages, knownExtensions )
+import Text.EditDistance ( defaultEditCosts, levenshteinDistance )
+
+showUnsupportedExtension :: Extension -> String
+showUnsupportedExtension (EnableExtension knownExt) = prettyShow knownExt ++ "which is not supported"
+showUnsupportedExtension extension@(DisableExtension _) = prettyShow extension ++ "which is not supported"
```

Comment:
maybe format it in the same way as `showUnsupportedLanguage`?

Sorry for nitpicking on minor things.

Reply:
Ah yeah, that makes sense.

Reply:
Fixed.